src/queries/text.ts

```diff
@@ -58,37 +62,66 @@ const getChildrenAsText = (
 const getNodeByText = (
   node: ReactTestInstance,
   text: TextMatch,
+  TextComponent: React.ComponentType,
   options: TextMatchOptions = {}
 ) => {
-  try {
-    const { Text } = require('react-native');
-    const isTextComponent = filterNodeByType(node, Text);
-    if (isTextComponent) {
-      const textChildren = getChildrenAsText(node.props.children, Text);
-      if (textChildren) {
-        const textToTest = textChildren.join('');
-        const { exact, normalizer } = options;
-        return matches(text, textToTest, normalizer, exact);
-      }
+  const isTextComponent = filterNodeByType(node, TextComponent);
+  if (isTextComponent) {
+    const textChildren = getChildrenAsText(node.props.children, TextComponent);
+    if (textChildren) {
+      const textToTest = textChildren.join('');
+      const { exact, normalizer } = options;
+      return matches(text, textToTest, normalizer, exact);
     }
-    return false;
-  } catch (error) {
-    throw createLibraryNotSupportedError(error);
   }
+  return false;
 };
 
+function isHostTextElement(
```

Comment:
Let's move that to `component-tree.ts` or `react-native-api.ts` (not existis yet but we have `react-native-api.test.tsx`)

Reply:
Looking at the body of the fn this could be called `isHostElementForType`, as the current function can be called with like `isHostElementForType(element, View)` and still makes sense and is useful.

BTW it seems that we could reuse `getCompositeParentOfType` for implementation:
`isHostElementForType(element, Component) {
    return getCompositeParentOfType(element, Component) != null;
 }